src/main/java/org/ow2/proactive/catalog/service/CatalogObjectGrantService.java

```diff
@@ -338,14 +347,19 @@ public CatalogObjectGrantMetadata updateCatalogObjectGrantForAGroup(String userG
         List<String> bucketsName = new LinkedList<>();
         bucketsName.add(bucketName);
         long bucketId = bucketRepository.findOneByBucketName(bucketName).getId();
-        long catalogObjectId = catalogObjectRevisionRepository.findDefaultCatalogObjectByNameInBucket(bucketsName,
-                                                                                                      catalogObjectName)
-                                                              .getId();
-        return catalogObjectGrantRepository.findCatalogObjectGrantEntitiesByBucketEntityIdAndCatalogObjectRevisionEntityId(bucketId,
-                                                                                                                           catalogObjectId)
-                                           .stream()
-                                           .map(CatalogObjectGrantMetadata::new)
-                                           .collect(Collectors.toList());
+        CatalogObjectRevisionEntity object = catalogObjectRevisionRepository.findDefaultCatalogObjectByNameInBucket(bucketsName,
+                                                                                                                    catalogObjectName);
+        if (object != null) {
+            long catalogObjectId = object.getId();
+            return catalogObjectGrantRepository.findCatalogObjectGrantEntitiesByBucketEntityIdAndCatalogObjectRevisionEntityId(bucketId,
+                                                                                                                               catalogObjectId)
+                                               .stream()
+                                               .map(CatalogObjectGrantMetadata::new)
+                                               .collect(Collectors.toList());
+        } else {
+            throw new DataIntegrityViolationException("Object: " + catalogObjectName + " was not found in bucket: " +
```

Comment:
In my opinion this exception should not be manually thrown:

- This is a Spring exception thrown when querying a DB
- The name does not represent the problem
- It will always return a http 500